module1-introduction-to-backend/r2-http-and-rest/README.md

```diff
@@ -0,0 +1,99 @@
+3. The browser initiates a TCP connection with the server.
+4. The browser sends an HTTP request to the server.
+5. The server handles the incoming request and sends an HTTP response.
+6. The browser displays the HTML content.
+
+All these steps happen each time we enter any URL and they happen in the background within milliseconds.
+
+Suppose you want to check out the upcoming coding bootcamps at http://www.re-coded.com. After you type the URL into your browser, your browser will extract the http part and recognize that it is the name of the network protocol to use. Then, it takes the domain name from the URL, in this case “re-coded.com”, and asks the Domain Name Server (DNS) to return an Internet Protocol (IP) address. Once the client knows the destination’s IP address, it then opens a connection to the server at that address, using the http protocol as specified. It will initiate a GET request to the server which contains the IP address of the host and optionally a data payload. The GET request contains the following text:
+```
+GET / HTTP/1.1
+Host: www.re-coded.com
+```
+This identifies the type of request "GET", the path on www.re-coded.com (in this case, “/“) and the protocol “HTTP/1.1.” It could also be "HTTP/2.0". These are latest revisions of the original HTTP.
+
+If the server is able to locate the path requested, the server might respond with the header:
+```
+HTTP/1.1 200 OK
+Content-Type: text/html
+```
+This header is followed by the content requested, which in this case is the information needed to render www.re-coded.com.
+
+The first line of the header, HTTP/1.1 200 OK, is confirmation that the server understands the protocol that the client wants to communicate with (HTTP/1.1), and an HTTP status code signifying that the resource was found on the server. The second line, Content-Type: text/html, shows the type of content that it will be sending to the client.
+
+If the server is not able to locate the path requested by the client, it will respond with the header:
+```
+HTTP/1.1 404 NOT FOUND
+```
+In this case, the server identifies that it understands the HTTP protocol, but the 404 NOT FOUND status code signifies that the specific piece of content requested was not found. This might happen if the content was moved or if you typed in the URL path incorrectly or if the page was removed.
+
+You can also try to explore this on your browser. If you're using Google Chrome, open a new tab, right click and select "Inspect". This opens up the developer console of the browser. From the top menu on this console, select "Network". Now type https://www.re-coded.com/ in your browser bar and hit Enter. You will see a bunch of requests processed in the Network tab as the website loads. If you navigate to the top most request, you should be able to see the request and response headers we just talked about.<br/><br/><img src="https://drive.google.com/uc?export=view&id=1SJPdV6-75u8iTr5xrlMGGb5IgGeH9nn9"><br/><br/>
```

Comment:
Sorry I was testing something with line breaks, I will remove these. Also I'm using the html tag because I want to reduce the display width of some of these images. Will add these changes.